util/log/src/main/scala/sbt/ConsoleLogger.scala

```diff
@@ -35,6 +35,19 @@ object ConsoleLogger {
   def isEscapeTerminator(c: Char): Boolean =
     c >= '@' && c <= '~'
 
+  /**
+   * Test if the character AFTER an ESC is the ANSI CSI, which means more than two characters until terminatioin.
```

Comment:
Right.  We never parsed that before, at all.   I also don't think we use it.

This fixes the initial bug and invalid handling if ESC[. We need to improve this anyway for the request about filtering color escapes separately from cursor control.   So I think this PR is ok as is.